### ISSUE TITLE
[vcpkg] Passing --dgml should suppress warnings (on stdout)

### ISSUE THREAD

**Issue by SunBlack** [microsoft/vcpkg#30180]
**Describe the bug**

Generated dgml files are invalid

![image](https://user-images.githubusercontent.com/483922/225008888-4c686a06-886a-4c1c-98b8-e55d53b23d5e.png)

**Environment**
- OS: Windows

**To Reproduce**
1. `./vcpkg depend-info boost --dgml > boost.dgml`
2. Open generated file with MSVC

**Expected behavior**

MSVC can show the graph

**Additional context**

The generated files looks currently like:

```xml
warning: Starting with the September 2023 release, the default triplet for vcpkg libraries will change from x86-windows to the detected host triplet (x64-windows). To resolve this message, add --triplet x86-windows to keep the same behavior.
<?xml version="1.0" encoding="utf-8"?><DirectedGraph xmlns="http://schemas.microsoft.com/vs/2009/dgml">...</DirectedGraph>
```

So the issue should exists since https://github.com/microsoft/vcpkg-tool/pull/881

**Possible solutions**

-  Print warnings on `stderr` instead of  `stdout`, as it is currently not possible to redirect just the xml file
- Change the parameter `--dgml` to require a file name (not sure if there is an use case to pipe the output directly to an tool)
- Suppress all warnings when passing `--dgml` and just print them in case the command itself failed

**Comment by autoantwort** [comment]
Related #33444

**Comment by SunBlack** [comment]
Fixed in microsoft/vcpkg-tool#1175.
